src/task.c

```diff
@@ -395,6 +325,154 @@ task_thread_create(const char            *name,
    return STATUS_OK;
 }
 
+/******************************************************************************
+ * Registering/deregistering threads that were not started by the task
+ * system.
+ ******************************************************************************/
+
+/*
+ * task_register_thread(): Register this new thread with the task system.
+ *
+ * This is for use by threads created outside of the splinter system.
+ *
+ * Registration implies:
+ *  - Acquire a new thread ID (index) for this to-be-active thread
+ *  - Allocate scratch space for this thread, and track this space..
+ */
+platform_status
+task_register_thread(task_system *ts,
+                     uint64       scratch_size,
+                     const char  *file,
+                     const int    lineno,
+                     const char  *func)
+{
+   threadid thread_tid;
+
+   debug_code(thread_tid = platform_get_tid());
+   debug_assert(
```

Comment:
Can we not make this a `platform_assert()`, so that any integrations using release binary will also get this check?

Thread registration is a one-time event in the life of a thread so, I think, the overhead of an additional assertion check (which this code adds in debug-builds) is very minor.

While building task-system unit-tests, and also during other integration, I found this business of registration needs to be done carefully. It's better to tighten this check for release builds, too,  as a safeguard.